app/jobs/run_submission_job.rb

```diff
@@ -6,19 +6,28 @@ class RunSubmissionJob < ApplicationJob
   def perform(submission_id)
     submission        = Submission.find_by_id!(submission_id)
     submission_result = submission.submission_results.create!
+    problem           = submission.problem
 
     runner = SubmissionRunners.runner_for_submission(submission)
     runner.call
 
     if runner.run_succeeded?
-      submission.status = "passed"
+      # FIXME: is this ok in general?
```

Comment:
Probably not long-term;  There are configurable whitespace rules that can be applied on a per problem basis, so a one-size fit's all approach here isn't right -- probably a strategy selector is necessary